client/src/leap/soledad/client/api.py

```diff
@@ -911,6 +931,35 @@ def raw_sqlcipher_operation(self, *args, **kw):
         """
         return self._dbpool.runOperation(*args, **kw)
 
+    #
+    # Service authentication
+    #
+
+    @inlineCallbacks
+    def get_or_create_service_token(self, service):
+        """
+        Return the stored token for a given service, or generates and stores a
+        random one if it does not exist.
+
+        These tokens can be used to authenticate services.
+        """
+        yield self.create_index('by-servicetoken', 'type', 'service')
```

Comment:
Do we need to sync the tokens between devices? I have the feeling that the token should be per device, but I'm not sure if I have good reasons for that.